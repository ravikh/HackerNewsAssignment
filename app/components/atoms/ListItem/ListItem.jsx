import React from 'react';
import ListItem from './ListItem.style';
 
export default ({children, className, type='ul'}) => <ListItem className={className} type={type}>{children}</ListItem>
