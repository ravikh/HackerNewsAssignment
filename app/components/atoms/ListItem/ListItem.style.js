import styled from 'styled-components'

const ListItem = styled.li`
  margin: 0;
  padding: 0.5rem 2rem;
  position: relative;

  &:nth-child(even) {
      background: #CCC;
  }

  ${(props) => (props.type === 'ol' && 
    `
      counter-increment: li;
      &::before {
        content: counter(li) ". ";
        display: inline-block;
        position: absolute;
        font-size: 16px;
        top: 0.5rem;
        left: 1.5rem;
      }
      @media ${props.theme.deviceBreakpoints.laptop} { 
        padding: 0.5rem 2rem;
      }
    `
    )}
    
`;

export default ListItem;