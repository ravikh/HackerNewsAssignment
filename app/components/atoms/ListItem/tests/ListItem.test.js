import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import ListItem from '../ListItem';

import Theme from '../../../../theme';

describe('ListItem', () => {
  let listItem = '';
  const theme = { ...Theme };

  test('should render correctly', () => {
    listItem = shallow(<ListItem theme={theme}>list item</ListItem>);
    expect(listItem).toMatchSnapshot();
  });
});