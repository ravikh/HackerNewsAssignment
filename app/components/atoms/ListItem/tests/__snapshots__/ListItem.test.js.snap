// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render correctly 1`] = `
<styled.li
  type="ul"
>
  list item
</styled.li>
`;
